generators/lib/solution-explorer.js

```diff
@@ -0,0 +1,115 @@
+var traverse = require('traverse');
+var jsonpath = require('jsonpath');
+const tabdownParser = require('tabdown-kfatehi');
+const vsParser = require('./vs-parse/index');
+
+class SolutionFileExplorer {
+  constructor(content) {
+    this.content = content || {};
+    this.solution = {};
+  }
+
+  toString() {
+    return traverse(this.solution)
+      .reduce(function (acc, x) {
+        var hasValue = x && x.value;
+        var indent = (this.level - 2) / 2;
+        return hasValue && acc.push({ indent, value: x.value.trim() }), acc;
+      }, [])
+      .map((value) => `${'\t'.repeat(value.indent)}${value.value}`)
+      .join('\r\n');
+  }
+
+  findNode(keyword) { 
+    const result = jsonpath.query(this.solution, `$..children[?(@.value == "${keyword}" || @.value.match(/^.*${keyword}/g))]`);
+    if(result && result.length) {
+      return result[0];
+    }
+  }
+
+  insert(node, ...values) {
+    if (node) {
+      const index = this.solution.children.indexOf(node);
+      this.solution.children.splice(index + 1, 0, ...values);
+    }
+  }
+
+  getProjectNodes(projects) {
+    return projects
+    .map(project => [
+      ({ value: `Project("{${project.projectTypeId.toUpperCase()}}") = "${project.name}", "${project.relativePath}", "{${project.id.toUpperCase()}}"`}), 
+      ({ value: `EndProject`})
+    ])
+    .reduce((a, b) => a.concat(b));
+  }
+
+  addNestedProject(node, project) {
+    node.children = node.children || [];
+    const { children, id } = project;
+
+    const nestedProjectsLines = children.map(child => {
+      if (child.children && child.children.length) {
+        this.addNestedProject(node, child);
+      }
+      return ({ value: `{${child.id.toUpperCase()}} = {${id.toUpperCase()}}` });
+    });
+
+    node.children.push(...nestedProjectsLines);
+  }
+
+  addProjectConfigurationPlatform(node, projects, configurations) {
+    if (node) {
+      node.children = node.children || [];
+
+      projects.forEach(project => 
```

Comment:
You can prefer functional style over procedure, so that this can be presented as

project.reduce((p,acc) => [...acc, configs.map(c => {c, p}], []);
node.children.push(...newItems)